core/src/main/java/org/elasticsearch/bootstrap/BootstrapCheck.java

```diff
@@ -135,9 +137,10 @@ static boolean enforceLimits(final Settings settings) {
         /**
          * test if the node fails the check
          *
+         * @param settings current node settings, which may influence the check
          * @return true if the node failed the check
          */
-        boolean check();
+        boolean check(Settings settings);
```

Comment:
I don't like that settings need to be passed into every check method. I'd prefer the specific setting needed be passed into `MinMasterNodesCheck` when it's constructed (look at the `MlockallCheck`).